src/gwc/src/main/java/org/geoserver/gwc/dispatch/GwcServiceProxy.java

```diff
@@ -173,11 +181,28 @@ public void verifyAccess(HttpServletRequest rawRequest) throws org.geotools.ows.
         } else if (rawRequest.getPathInfo().toLowerCase().startsWith("/service/gmaps") || rawRequest.getPathInfo().toLowerCase().startsWith("/service/ve")) {
             String layerstr = (String) parameters.get("LAYERS");
                         
-            if (layerstr != null) {                
+            if (layerstr != null) {
+            	int level = Integer.parseInt((String)parameters.get("zoom"));
+            	long col = Long.parseLong((String) parameters.get("x"));
+            	long row = Long.parseLong((String) parameters.get("y"));
+        		try {
+					long [] converted = GMapsConverter.convert(level, col, row);
+					col = converted[0];
+					row = converted[1];
+				} catch (org.geowebcache.service.ServiceException e) {
+					throw new ServiceException(e);
+				}
+            	
                 String[] layers = layerstr.split(",");
                 for (String layerName: layers) {
                     layerName = layerName.trim();
-                    GWC.get().verifyAccessLayer(layerName, null);
+                	GeoServerTileLayer layer = (GeoServerTileLayer) GWC.get().getTileLayerByName(layerName);
+                	String srs = "EPSG:3857";
+                	GridSubset gridSubset =  layer.getGridSubset(srs);
+                	if(gridSubset == null){
+                		srs = "EPSG:900913";
+                	}
+                    GWC.get().verifyAccessTiledLayer(layerName ,srs, level, col, row);
```

Comment:
@jodygarnett I know about 2 other places for the kml and tms requests direct above which I'll try to fix. I'll have a look for more places to fix. I think a check on the call sources of the ```verifyAccessLayer``` should point that out?